components/family-studies/api/src/main/java/org/phenotips/studies/family/internal/PhenotipsFamily.java

```diff
@@ -201,7 +211,7 @@ public String getURL(String actions)
     public String getWarningMessage()
     {
         BaseObject familyObject = getXDocument().getXObject(Family.CLASS_REFERENCE);
-        return familyObject.getIntValue(WARNING) == 0
+        return familyObject.getIntValue(FIELD_HAS_SENSITIVE_DATA) == FIELD_VALUE_CHECKED
```

Comment:
The logic changed here. You used to check if the "has sensitive data" box is unchecked, now you're checking if it is checked. For this reason, the message sent to the frontend is always an empty string and the generic message is displayed to the user.

```suggestion
        return familyObject.getIntValue(FIELD_HAS_SENSITIVE_DATA) != FIELD_VALUE_CHECKED
```